src/Citeproc/Eval.hs

```diff
@@ -1142,16 +1148,41 @@ evalItem layout (position, item) = do
            , contextPosition = position
            },
         st{ stateReference = ref
-          , stateUsedYearSuffix = False }))
+          , stateUsedYearSuffix = False
+          , stateUsedIdentifier = False
+          , stateUsedTitle = False
+          }))
         $ do xs <- mconcat <$> mapM eElement (layoutElements layout)
+
+             -- find identifiers that can be used to hyperlink the title 
+             let mbident = 
+                    foldl (<|>) Nothing
+                      [ IdentDOI   <$> (valToText =<< lookupVariable "DOI" ref)
+                      , IdentPMCID <$> (valToText =<< lookupVariable "PMCID" ref)
+                      , IdentPMID  <$> (valToText =<< lookupVariable "PMID" ref)
+                      , IdentURL   <$> (valToText =<< lookupVariable "URL" ref)
+                      ]
+             -- hyperlink any titles in the output
+             let linkTitle target (Tagged TagTitle x) = Tagged (TagLink target) x
+                 linkTitle _ x = x
+             
+             -- when no links were rendered, hyperlink title, if it exists
+             usedLink <- gets stateUsedIdentifier
+
+             let xs' = case mbident of
+                         Nothing     -> xs
+                         Just ident  -> if usedLink
+                                           then xs
+                                           else fmap (transform (linkTitle ident)) xs
```

Comment:
Currently `transform` is called twice -- one for `linkTitle` and one for `addLangtoFormatting`.  Should these two uses be merged into one?

Reply:
If it can be done without contorting the code, this seems sensible.